### PR TITLE
Simplify vagrant setup for backend AND frontend development

### PR DESCRIPTION
**Pull request by hennevogel**
* Both parts of the app now can be started in development mode
 * vagrant exec contrib/start_development_backend
 * vagrant exec rails server
* Development and testing is completely separated
  * vagrant exec script/start_development_backend
  * vagrant exec contrib/start_test_backend
* No messing with test & fixtures anymore. Just do what all rails devs do
  * rake db:fixtures:load
* Set OBS_VAGRANT_MEM/OBS_VAGRANT_CPU in your ~/.profile to have more ram/cpus in VirtualBox
* switched base box to the official Leap 42.1 box

### CODE REVIEW

**Comment by hennevogel** [comment]
@ChrisBr @mdeniz @bgeuken I guess you want to test this branch before we merge it :-)

**Comment by lethliel** [comment]
@hennevogel works for me

**Comment by hennevogel** [comment]
One pitfall for 'migrating' from the old setup is that you should make sure that your `src/api/config/options.yml` and `src/api/config/database.yml` work with the new backend, the new setup is made to work with the `.example` files. Especially the `source_port` in `options.yml` should be **5352**

File: contrib/common.sh [hunk 1]
```diff
@@ -86,60 +86,10 @@ function configure_app() {
...
-  TMP_IN_FSTAB=$(grep "$MOUNT_DIR" /etc/fstab)
-  if [ -z "$TMP_IN_FSTAB" ];then
-    echo " - Adding $TMP_DIR to fstab"
-    echo -e "$TMP_DIR $MOUNT_DIR none bind 0 0" >> /etc/fstab
-  fi
-
+  copy_example_file database.yml && \
+  rake -f /vagrant/src/api/Rakefile db:version || \
+  rake -f /vagrant/src/api/Rakefile db:create && \
+  rake -f /vagrant/src/api/Rakefile db:setup
```
Comment (bgeuken) [inline contrib/common.sh]: I would indent the last 3 lines to make it more obvious that the belong together

**Comment by bgeuken** [comment]
Works like a charm.

@lethliel @hennevogel Good Job!

**Comment by ChrisBr** [comment]
LGTM too :+1:

**Comment by hennevogel** [comment]
Then let's go

**Pull request merged**